### ISSUE TITLE
The location of definitions of and proofs about specific data relations

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#173]
I've been thinking about this for a couple of months but #172 has forced the issue.

When we added `All₂` to `Data.Vec.All`, we entirely duplicated the functionality of `Pointwise'` in `Relation.Binary.Vec.Pointwise`. Likewise adding `All₂` to `Data.List.All` will duplicate the functionality of `Pointwise` in `Relation.Binary.List.Pointwise`.

We shouldn't encourage duplicated functionality, so the simple solution is to simply say use `Pointwise` instead. The problem with this is that it would be useful to add proofs of how these relations interact with `List`/`Vec` operations. e.g.
```agda
pw-tabulate⁺ : ∀ {a b ℓ} {A : Set a} {B : Set b} {_~_ : REL A B ℓ} →
                     ∀ {n} {f : Fin n → A} {g : Fin n → B} → (∀ i → f i ~ g i) →
                     Rel _~_ (tabulate f) (tabulate g)
pw-tabulate⁺ {n = zero} f~g  = []
pw-tabulate⁺ {n = suc n} f~g = (f~g fzero) ∷ pw-tabulate⁺ (f~g ∘ fsuc)
```
It seems odd to be adding these inside the `Relation.Binary` subtree and not inside the `Data.List` subtree. In fact it seems strange to me to have the definitions inside the `Relation.Binary` at all. If we were being consistent then, for example, `Data.List.Any` should be in `Relation.Unary.List.Any`. The same issue applies to not just to `List` but the other data types `Vec`, `Sigma` and `Product` as well.

I think we have two options:

1. Leave it as is, and add proofs about `List` operations to the `Relation.Binary.List.Pointwise` file
2. Move the contents of `Relation.Binary.List` to `Data.List.Relation`. By leaving the original files in place, and re-exporting from the new location we could make this move entirely backwards compatible.

Personally I think the 2nd is a lot tidier and less confusing than the 1st. Thoughts?

**Comment by strake** [comment]
I also prefer option 2 and modified my PR to make it so.

**Comment by andreasabel** [comment]
Option 2. sound preferable.  In fact, I programmed relations like All2 myself because I would not think to search for them in the Relation subtree.  I extrapolated from Data.List.All, thus, did not find anything.

**Comment by MatthewDaggitt** [comment]
> Option 2. sound preferable. In fact, I programmed relations like All2 myself because I would not think to search for them in the Relation subtree. I extrapolated from Data.List.All, thus, did not find anything.

Seems like we're in agreement. @gallais any thoughts?

Otherwise I'll go through and make the changes at some point this week.

**Comment by gallais** [comment]
I too like Option 2 better.

**Comment by strake** [comment]
Should we also deprecate `Data.Vec.All.All₂` or make it an alias of `Relation.Binary.Vec.Pointwise.Pointwise′`?

**Comment by MatthewDaggitt** [comment]
Yes, we do need to deprecate it. When I get a moment I'll redirect it to "Data.Vec.Relation.Pointwise"...

**Comment by MatthewDaggitt** [comment]
Closed with 2e52440
